src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs

```diff
@@ -302,22 +301,21 @@ void SetupPagingFromQueryMetadata()
             {
                 if (result.Metadata.Query.Offset != null)
                 {
-                    var offset = (int)QueryBuilderHelper.GetLongValue(result.Metadata.Query, result.Metadata, result.QueryParameters, result.Metadata.Query.Offset, 0);
+                    var offset = QueryBuilderHelper.GetLongValue(result.Metadata.Query, result.Metadata, result.QueryParameters, result.Metadata.Query.Offset, 0);
                     result.Offset = offset;
                     result.Start = start + offset;
                 }
 
                 if (result.Metadata.Query.Limit != null)
                 {
-                    pageSize = (int)QueryBuilderHelper.GetLongValue(result.Metadata.Query, result.Metadata, result.QueryParameters, result.Metadata.Query.Limit, int.MaxValue);
+                    pageSize = Math.Min(int.MaxValue, QueryBuilderHelper.GetLongValue(result.Metadata.Query, result.Metadata, result.QueryParameters, result.Metadata.Query.Limit, int.MaxValue));
```

Comment:
I think you also need to do the same below:

https://github.com/ravendb/ravendb/blob/8bd489213e85e994ee6e0648137172672eec7e27/src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs#L209